File: Medium/Subset Sum on Generated Sequence/subset-sum-on-generated-sequence.cpp
class Solution {
  public:
    bool isPossible(vector<int>& arr, int s, int x) {
        // code here
        vector<int> nums = {s};
        
        int total = s;
        
        for (int val : arr){
            int next = total + val;
            
            if (next <= x){
                nums.push_back(next);
            }
            
            total += next;
            
            if (total > x && next > x) break;
        }
        
        for (int i = nums.size() - 1 ; i >= 0 ; i--){
            if (nums[i] <= x){
                x -= nums[i];
            }
            
            if (x == 0) return true;
        }
        
        return false;
        
    }
};

File: Medium/Subset Sum on Generated Sequence/subset-sum-on-generated-sequence.py
class Solution:
    def isPossible(self, arr, s, x):
        # code here 
        finalSeq = [s]
        
        total = s
        
        for i in arr:
            next = total + i
            
            if next<=x:
                finalSeq.append(next)
                
            total += next
            
            if total>x and next>x:
                break
        
        
        for i in range(len(finalSeq)-1 ,-1 , -1):
            if finalSeq[i] <= x:
                x -= finalSeq[i]
            
            if x==0:return True
        
        return False
            
        

        
        